server/website/website/tasks/async_tasks.py

```diff
@@ -190,11 +190,14 @@ def aggregate_target_results(result_id, algorithm):
     # implement a sampling technique to generate new training data).
     newest_result = Result.objects.get(pk=result_id)
     has_pipeline_data = PipelineData.objects.filter(workload=newest_result.workload).exists()
-    if newest_result.session.tuning_session == 'lhs':
+    if not has_pipeline_data or newest_result.session.tuning_session == 'lhs':
+        if not has_pipeline_data and newest_result.session.tuning_session == 'tuning_session':
+            LOG.debug("Background tasks haven't ran for this workload yet, picking random data.")
+
         all_samples = JSONUtil.loads(newest_result.session.lhs_samples)
         if len(all_samples) == 0:
             knobs = SessionKnob.objects.get_knobs_for_session(newest_result.session)
-            all_samples = gen_lhs_samples(knobs, 100)
+            all_samples = gen_lhs_samples(knobs, 10)
```

Comment:
We should generate 100 samples if `newest_result.session.tuning_session == 'lhs'` or 10 samples if `not has_pipeline_data`